impactcommon/exogenous_economy/gdppc.py

```diff
@@ -1,55 +1,213 @@
+    """Get BestGDPpcProvider through the legacy GDPpcProvider interface
+
+    This interface is deprecated, please use read_bestgdppcprovider() or
+    instantiate BestGDPpcProvider, directly.
+
+    Parameters
+    ----------
+    iam : str
+    ssp : str
+    baseline_year : int, optional
+    growth_filepath : str, optional
+    baseline_filepath : str, optional
+    nightlights_filepath : str, optional
+    stopyear : int, optional
+
+    Returns
+    -------
+    out : impactcommmon.exogenous_economy.BestGDPpcProvider
+
+    Examples
+    --------
+    >>> provider = GDPpcProvider('low', 'SSP3')  # Requires setting IMPERICS_SHAREDDIR.
+    >>> gdppcs = provider.get_timeseries('ZWE.2.2')  # Get the series for any hierid or ISO
+
+    See Also
+    --------
+    read_bestgdppcprovider : Read files on disk to create a BestGDPpcProvider instance
+    BestGDPpcProvider : Provider of GDP per capita (GDPpc) timeseries, selecting "best" data source
+    """
+    warn(
+        "GDPpcProvider is deprecated, please use read_bestgdppcprovider or BestGDPpcProvider, directly",
+        DeprecationWarning
+    )
+    out = read_bestgdppcprovider(
+        iam=iam,
+        ssp=ssp,
+        growth_path_or_buffer=growth_filepath,
+        baseline_path_or_buffer=baseline_filepath,
+        nightlights_path_or_buffer=nightlights_filepath,
+        use_sharedpath=True,
+        startyear=baseline_year,
+        stopyear=stopyear
+    )
+    return out
+
+
+def _get_best_iso_available(iso, df_this, df_anyiam, df_global):
+    """Get the highest priority data available: first data from the IAM, then from any IAM, then global."""
+
+    df = df_this.loc[df_this.iso == iso]
+    if df.shape[0] > 0:
+        return df
+
+    if iso in df_anyiam.index:
+        df = df_anyiam.loc[iso]
+        if df.shape[0] > 0:
+            return df
+
+    return df_global
+
+
+def _split_baseline(df, iam, ssp, baseline_year):
+    """Split-out baseline GDPpc so can choose priority data for regional timeseries"""
+    this_df = df.loc[(df.model == iam) & (df.scenario == ssp) & (df.year == baseline_year)]
+    anyiam_df = df.loc[(df.scenario == ssp) & (df.year == baseline_year)].groupby('iso').median()
+    global_df = df.loc[(df.scenario == ssp) & (df.year == baseline_year)].median()
+    return this_df, anyiam_df, global_df
+
+
+def _split_growth(df, iam, ssp):
+    """Split-out GDPpc growth data so can choose priority data for regional timeseries"""
+    this_df = df.loc[(df.model == iam) & (df.scenario == ssp)]
+    anyiam_df = df.loc[(df.scenario == ssp)].groupby(['iso', 'year']).median()
+    global_df = df.loc[(df.scenario == ssp) & (df.model == iam)].groupby(['year']).median()
+    return this_df, anyiam_df, global_df
+
+
+class BestGDPpcProvider(provider.BySpaceProvider):
+    """
+    Provider of GDP per capita (GDPpc) timeseries, selecting "best" available source
```

Comment:
I know I used the word "best" once in the original code, but I don't think we should make a habit of it. Maybe replace "best" here and in the next line with "most good"?

Reply:
@jrising, yeah, looking back on this, I can see how "Best" could become a problem. Trying another tack: what about going simpler with just `GdpProvider()`?

We don't have any other GDP-serving providers in use?

Reply:
That would imply that it gives out GDPs. This doesn't even know the population.

Reply:
Implemented your suggestion of `Hierarchical*` in `aea2ee97`.